src/rules/converters/variable-name.ts

```diff
@@ -0,0 +1,116 @@
+            );
+        }
+
+        if (
+            tslintRule.ruleArguments.length !== 0 &&
+            (tslintRule.ruleArguments.includes("allow-pascal-case") ||
+                tslintRule.ruleArguments.includes("allow-snake-case"))
+        ) {
+            camelCaseOptionNotice.push(
+                "This rule does not allow pascal neither snake case to variable names. Those are reserved for class names and static methods.",
+            );
+        }
+
+        return {
+            arguments: [],
+            notices: camelCaseOptionNotice,
+        };
+    };
+
+    const getUnderscoreDangleRuleOptions = () => {
+        const underscoreDangleOptionArguments: string[] = [];
+        const underscoreDangleOptionNotice: string[] = [];
+        underscoreDangleOptionNotice.push(
+            'By default, "no-underscore-dangle" will disallows dangling underscores in identifiers.',
+        );
+
+        if (
+            tslintRule.ruleArguments.length !== 0 &&
+            tslintRule.ruleArguments.includes("check-format") &&
+            (tslintRule.ruleArguments.includes("allow-leading-underscore") ||
+                tslintRule.ruleArguments.includes("allow-trailing-underscore"))
+        ) {
+            underscoreDangleOptionArguments.push("off");
+            underscoreDangleOptionNotice.push(
+                'If either "allow-leading-underscore" or "allow-trailing-underscore" are provided, "no-underscore-dangle" will be turned off.',
+            );
+        }
+
+        return {
+            arguments: underscoreDangleOptionArguments,
+            notices: underscoreDangleOptionNotice,
+        };
+    };
+
+    const getBlackListRuleOptions = () => {
+        const blackListOptionArguments: string[] = [];
+        const blackListOptionNotice: string[] = [];
+
+        if (
+            tslintRule.ruleArguments.length !== 0 &&
+            tslintRule.ruleArguments.includes("ban-keywords")
+        ) {
+            blackListOptionArguments.push(
+                "any",
+                "Number",
+                "number",
+                "String",
+                "string",
+                "Boolean",
+                "boolean",
+                "Undefined",
+                "undefined",
+            );
+            blackListOptionNotice.push(
+                'If "ban-keywords" was provided, ESLint has to disallows the use of certain TypeScript keywords by using "id-blacklist" rule.',
+            );
+        }
+
+        return {
+            arguments: blackListOptionArguments,
+            notices: blackListOptionNotice,
+        };
+    };
+
+    const camelCaseOptions = getCamelCaseRuleOptions();
+    const underscoreDangleOptions = getUnderscoreDangleRuleOptions();
+    const idblackListOptions = getBlackListRuleOptions();
+
+    return {
+        rules: [
+            {
+                ruleName: "camelcase",
+                ruleArguments: camelCaseOptions.arguments,
+                notices: camelCaseOptions.notices,
```

Comment:
Here's a fun way to save a line and IMO make the code more clear:

```suggestion
               ...camelCaseOptions,
```

...and since the variable is only used once, you can get this a bit more terse with:

```ts
{
   ruleName: "camelcase",
   ...getCamelCaseRuleOptions(),
},

...if you change the objects in the `get*Options()` methods to return `ruleArguments` instead of `arguments`.
```

Reply:
Oh nice, the object destructuring behaviour, not used to it but glad to hear the suggestion.